test/integration/consul-container/test/wanfed/acl_bootstrap_replication_test.go

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package wanfed
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	libcluster "github.com/hashicorp/consul/test/integration/consul-container/libs/cluster"
+)
+
+const replicationPolicyRules = `
+acl      = "write"
+operator = "write"
+
+service_prefix "" {
+  policy     = "read"
+  intentions = "read"
+}
+`
+
+var retryFuncTimer = &retry.Timer{
+	Timeout: 60 * time.Second,
+	Wait:    2 * time.Second,
+}
+
+// retryFunc is a helper to retry the given function until it returns a nil
+// error. It returns the value from the function on success.
+func retryFunc[T any](t *testing.T, f func() (T, error)) T {
+	var result T
+	retry.RunWith(retryFuncTimer, t, func(r *retry.R) {
+		val, err := f()
+		require.NoError(r, err)
+		result = val
+	})
+	return result
+}
```

Comment:
I added this to try to declutter the retry loops in the test case a little. One gotcha is that the `retry.R` is not available to the provided function (`f`), so the caller shouldn't check assertions.